lib/sqlalchemy/sql/schema.py

```diff
@@ -385,6 +385,11 @@ def foreign_keys(self) -> Set[ForeignKey]:
 
     """
 
+    periods: DictCollection[Period]
```

Comment:
There's a `Properties` class that already does what `DictCollection` does, we should use that.   https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/util/_collections.py#L170

As it looks like we have an idea that "table.periods.<name>" is going to be part of an API here, DML features seem like there would at least be something that represents for example "FOR PORTION OF"

     stmt = update(table).for_portion_of(table.p.date_period, date1, date2)

something like that?   "for portion of" seems like part of the SQL spec (which I dont have for the version that includes this)